### DISCUSSION TITLE
Prebuilt Binary Support for Node.js v22.x

### DISCUSSION THREAD

**Discussion by Jelmerovereem** [Project-OSRM/osrm-backend#7068] [Q&A] [answered]
We want to update our Node.js version from v18 to v22, but got an error during the npm install process. The error message says that OSRM doesn't have prebuilt binaries for Node ABI version 127 (Node.js v22).
Error:

```bash
node-pre-gyp ERR! stack Error: response status 404 Not Found on https://github.com/Project-OSRM/osrm-backend/releases/download/v5.27.1/node_osrm-v5.27.1-node-v127-linux-x64-Release.tar.gz
```

Are there plans to support Node.js v22 and release prebuilt binaries for ABI v127?
If building from source is the only option, I would like some guidance or next steps to achieve this.

Using `npm install --build-from-source` gives me a lot of `make` errors.

Thank you!

**Comment by hsource** [comment]
Looks like it's coming in a few weeks: https://github.com/Project-OSRM/osrm-backend/issues/7091#issuecomment-2733398444

If you can't wait, we built a version from the pre-release code. You can install it too for the time being.

```
yarn add wanderlog/osrm-backend#v5.28.0-wanderlog.1
```

**Comment by DennisOSRM** [comment, marked as answer]
The latest release v26.4.0 just dropped and supports node 22/24/25.
